apps/arweave/src/ar_nonce_limiter_server.erl

```diff
@@ -109,12 +113,27 @@ terminate(_Reason, _State) ->
 %%%===================================================================
 %%% Private functions.
 %%%===================================================================
+
 make_nonce_limiter_update(_SessionKey, not_found, _IsPartial) ->
 	not_found;
 make_nonce_limiter_update(SessionKey, Session, IsPartial) ->
 	StepNumber = Session#vdf_session.step_number,
 	Checkpoints = maps:get(StepNumber, Session#vdf_session.step_checkpoints_map, []),
+	StepCheckpointsMap = Session#vdf_session.step_checkpoints_map,
 	%% Clear the step_checkpoints_map to cut down on the amount of data pushed to each client.
+	RecentStepNumbers = get_recent_step_numbers(StepNumber),
+	StepCheckpointsMap2 = maps:with(RecentStepNumbers, StepCheckpointsMap),
 	#nonce_limiter_update{ session_key = SessionKey,
 			is_partial = IsPartial, checkpoints = Checkpoints,
-			session = Session#vdf_session{ step_checkpoints_map = #{} } }.
+			session = Session#vdf_session{ step_checkpoints_map = StepCheckpointsMap2 } }.
```

Comment:
See [comment below](https://github.com/ArweaveTeam/arweave/pull/577/files#r1612193904)

We can (eventually) remove the `checkpoints` field, right? Since the data it sores is redundant to `session#vdf_session.step_checkpoints_map`